github/repos_releases.go

```diff
@@ -488,3 +488,55 @@ func (s *RepositoriesService) UploadReleaseAsset(ctx context.Context, owner, rep
 	}
 	return asset, resp, nil
 }
+
+// UploadReleaseAssetFromRelease uploads an asset using the UploadURL that's embedded
+// in a RepositoryRelease object.
+//
+// This is a convenience wrapper that extracts the release.UploadURL (which is usually
+// templated like "https://uploads.github.com/.../assets{?name,label}") and uploads
+// the provided data (reader + size) using the existing upload helpers.
+//
+// GitHub API docs: https://docs.github.com/rest/releases/assets#upload-a-release-asset
+//
+//meta:operation POST /repos/{owner}/{repo}/releases/{release_id}/assets
+func (s *RepositoriesService) UploadReleaseAssetFromRelease(ctx context.Context, release *RepositoryRelease, opts *UploadOptions, reader io.Reader, size int64) (*ReleaseAsset, *Response, error) {
```

Comment:
I don't understand why we need to pass the entire `RepositoryRelease` object if we can simply pass `uploadURL`. Like this:
```go
func (s *RepositoriesService) UploadReleaseAssetFromRelease(ctx context.Context, uploadURL string, opts *UploadOptions, reader io.Reader, size int64) (*ReleaseAsset, *Response, error) {
```